### DISCUSSION TITLE
After launching lazer once on Windows, it's unable to be re-opened until a full restart.

### DISCUSSION THREAD

**Discussion by zachmanthethird** [ppy/osu#16992] [Q&A] [answered]
Version 2022.217.0. While I don't personally have this issue, I've received countless complaints from tournament staff (streamers) that they cannot re-open lazer without restarting their entire computers. Finally got one of them to give me log files so that I can at least start the discussion.
Appears to boil down to this:
```2022-02-26 06:49:27 [error]: An unhandled error has occurred.
2022-02-26 06:49:27 [error]: System.IO.IOException: Access to the path 'C:\Users\marcs\AppData\Roaming\osu\tournaments\tc\videos' is denied.
2022-02-26 06:49:27 [error]: at System.IO.FileSystem.MoveDirectory(String sourceFullPath, String destFullPath)
2022-02-26 06:49:27 [error]: at System.IO.Directory.Move(String sourceDirName, String destDirName)
2022-02-26 06:49:27 [error]: at osu.Game.Tournament.IO.TournamentStorage.<>c__DisplayClass7_0.<ChangeTargetStorage>b__2()
2022-02-26 06:49:27 [error]: at osu.Game.IO.MigratableStorage.AttemptOperation(Action action, Int32 attempts, Boolean throwOnFailure)
2022-02-26 06:49:27 [error]: at osu.Game.Tournament.IO.TournamentStorage.ChangeTargetStorage(Storage newStorage)
2022-02-26 06:49:27 [error]: at osu.Game.Tournament.IO.TournamentStorage..ctor(Storage storage)
2022-02-26 06:49:27 [error]: at osu.Game.Tournament.TournamentGameBase.load(Storage baseStorage)
```
Full [runtime.log.txt](https://github.com/ppy/osu/files/8146073/runtime.log.txt)

**Comment by bdach** [comment]
If this is the case then it's likely that a previous instance of the tournament client hasn't exited cleanly.

Please ask the people that are seeing this to check task manager and see if the osu! process is still visible there. If it is, then people seeing this can click the process entry, click "Create dump file", and then send that via email to pe@ppy.sh. Do **NOT** post dump files publically as they may contain user credentials which could be compromised.

**Reply by peppy** [reply]
It's weird that it's attempting to move the files on startup though... requires some further investigation I think.

**Reply by bdach** [reply]
These moves [are only there for case-sensitive OSes](https://github.com/ppy/osu/blob/6eb79c37d88a42ce1800b4a5a79cac0610b87112/osu.Game.Tournament/IO/TournamentStorage.cs#L56-L66) and presumably do nothing on windows. They could be guarded with a platform check, I guess - but if the video decoder is hanging on and keeping video files open then both issues need to be fixed anyhow, I'd say.

**Reply by peppy** [reply]
Either that or the user is accessing the videos externally (potentially OBS) for other purposes. Quite feasible given the use case.

**Comment by peppy** [comment]
@zachmanthethird So two things:
- Can you have the user try to start the tournament mode without OBS or anything else running just once, so it can correctly migrate the files it's trying to?
- Can you have the user take a screenshot of their tournament folder? Specifically I'm interested in confirming whether the first character of the `flags` `mods` and `videos` folders are lower or upper case.

**Comment by zachmanthethird** [comment, marked as answer]
It does appear as though it's tied to OBS issue; however, we're still not certain why. Nothing is directly accessing the videos in the tournament folder...
However, that's clearly not a lazer problem. Thanks for the help :D

P.S. (for anyone finding this issue later) the folders were all uppercase (`Flags`, `Mods`, `Videos`) and the process was not in the task manager.

**Reply by peppy** [reply]
Does it still happen after you successfully ran once (if OBS is running)?

**Reply by zachmanthethird** [reply]
It doesn't anymore, but we've had to completely rebuild our streamer collection.
My best guess is there was a file that was actually a link back to the video in the tournaments folder, but I can't be sure.
Everything's working fine now :+1:

**Reply by peppy** [reply]
Good to know.

One thing to be aware of is that this issue will only come up if you use lower case folder names (as osu! will try to correct them). For future new tournaments you can make sure your folders are all capital first letter and you will never run into this issue, even if OBS is accessing files.
